hyperspy/signal_tools.py

```diff
@@ -238,7 +243,162 @@ def apply(self):
         self.span_selector_switch(on=True)
         self.last_calibration_stored = True
 
+@add_gui_method(toolkey="hyperspy.EELSSpectrum.print_edges_table")
+class EdgesRange(SpanSelectorInSignal1D):
+    units = t.Unicode()
+    edges_list = t.Tuple()
+    only_major = t.Bool()
+    order = t.Unicode('closest')
+    complementary = t.Bool(True)
+    
+    def __init__(self, signal):
+        super(EdgesRange, self).__init__(signal)
+        if signal.axes_manager.signal_dimension != 1:
+            raise SignalDimensionError(
+                signal.axes_manager.signal_dimension, 1)
+        self.units = self.axis.units
+        self.active_edges = []
+        self.active_complementary_edges = []
+        self.slp = SpectrumLabelPosition(self.signal)
+        self.btns = []
+
+        self._get_edges_info_within_energy_axis()
+
+    def _get_edges_info_within_energy_axis(self):
+        mid_energy = (self.axis.low_value + self.axis.high_value) / 2
+        rng = self.axis.high_value - self.axis.low_value 
+        self.edge_all = np.asarray(get_edges_near_energy(mid_energy, rng, 
+                                                         order=self.order))
+        info = get_info_from_edges(self.edge_all)
+        
+        energy_all = []
+        relevance_all = []
+        description_all = []
+        for d in info:
+            onset = d['onset_energy (eV)']
+            relevance = d['relevance']
+            threshold = d['threshold']
+            edge_ = d['edge']            
+            description = threshold + '. '*(threshold !='' and edge_ !='') + edge_
+        
+            energy_all.append(onset)
+            relevance_all.append(relevance)
+            description_all.append(description)
+        
+        self.energy_all = np.asarray(energy_all)
+        self.relevance_all = np.asarray(relevance_all)
+        self.description_all = np.asarray(description_all)
+
+    def update_table(self):
+        figure_changed = self.slp._check_signal_figure_changed()
+        if figure_changed:
+            self._clear_all_markers()
```

Comment:
When changing navigation index, drawing of the line is connected to the `axes_manager`: there should be a method which update the labels (their vertical position) when the nagivation indices change to take into account changes in intensity.
It may be worth looking at how some methods of the `BackgroundRemoval` tool are connected or how plotting x-rays lines works too.

Reply:
The event `axes_manager.events.indices_changed` is connected to a method `_on_nav_change` now so that whenever the indices of `axes_manager` changed, it gets the new figure properties and update the existing labels.